ctapipe/reco/tests/test_HillasReconstructor.py

```diff
@@ -72,87 +82,273 @@ def test_h_max_results():
     p2 = SkyCoord(alt=0 * u.deg, az=90 * u.deg, frame=horizon_frame)
     circle4 = HillasPlane(p1=p1, p2=p2, telescope_position=[-1, 0, 0] * u.m)
 
+    # Create a dummy subarray
+    # (not used here, but required to initialize the reconstructor)
+    subarray = SubarrayDescription(
+        "test array",
+        tel_positions={1: np.zeros(3) * u.m},
+        tel_descriptions={
+            1: TelescopeDescription.from_name(
+                optics_name="SST-ASTRI", camera_name="CHEC"
+            )
+        },
+    )
+
     # creating the fit class and setting the the great circle member
-    fit = HillasReconstructor()
-    fit.hillas_planes = {1: circle1, 2: circle2, 3: circle3, 4: circle4}
+    fit = HillasReconstructor(subarray)
+    hillas_planes = {1: circle1, 2: circle2, 3: circle3, 4: circle4}
 
     # performing the direction fit with the minimisation algorithm
     # and a seed that is perpendicular to the up direction
-    h_max_reco = fit.estimate_h_max()
+    h_max_reco = fit.estimate_h_max(hillas_planes)
     print("h max fit test minimise:", h_max_reco)
 
     # the results should be close to the direction straight up
     np.testing.assert_allclose(h_max_reco.value, 0, atol=1e-8)
     # np.testing.assert_allclose(fitted_core_position.value, [0, 0], atol=1e-3)
 
 
-def test_reconstruction():
+def test_parallel_reconstruction():
     """
-    a test of the complete fit procedure on one event including:
-    • tailcut cleaning
+    Test shower's reconstruction procedure:
+    • image cleaning
     • hillas parametrisation
     • HillasPlane creation
-    • direction fit
-    • position fit
+    • shower direction reconstruction in the sky
+    • shower core reconstruction in the ground
 
-    in the end, proper units in the output are asserted """
-    filename = get_dataset_path("gamma_test_large.simtel.gz")
+    Tested,
+    - starting from CameraFrame,
+    - starting from TelescopeFrame,
+
+    The test checks that the old approach (using CameraFrame) and the new one
+    (using TelescopeFrame) provide compatible results and that we are able to
+    reconstruct a positive number of events.
+    """
+
+    filename = get_dataset_path(
+        "gamma_LaPalma_baseline_20Zd_180Az_prod3b_test.simtel.gz"
+    )
 
     source = EventSource(filename, max_events=10)
     calib = CameraCalibrator(subarray=source.subarray)
 
-    horizon_frame = AltAz()
+    fit_CameraFrame = HillasReconstructor(source.subarray)
+    fit_TelescopeFrame = HillasReconstructor(source.subarray)
```

Comment:
Why do we need two reconstructors?

Reply:
To test that they give compatible results since they start from different frames

Reply:
But cant we use the same reconstructor for both frames?

Reply:
because when this code was done, event.dl2 didn't exist....

Reply:
fixed